From a0446ff36ae83bd1f1a1115bfc0268164a319ec2 Mon Sep 17 00:00:00 2001
From: Hans Christian Lonstad <hcl@datarespons.no>
Date: Sat, 2 Mar 2019 16:38:20 +0100
Subject: [PATCH] Set service dependency on polkit if used

If ModemManager starts without being able to get to polkit, operations
will fail.
---
 data/Makefile.am                    |  5 +++++
 data/ModemManager.service.polkit.in | 22 ++++++++++++++++++++++
 2 files changed, 27 insertions(+)
 create mode 100644 data/ModemManager.service.polkit.in

diff --git a/data/Makefile.am b/data/Makefile.am
index ee05bdf6..8081148e 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -28,9 +28,14 @@ systemdsystemunitdir = $(SYSTEMD_UNIT_DIR)
 systemdsystemunit_in_files = ModemManager.service.in
 if HAVE_SYSTEMD
 systemdsystemunit_DATA = ModemManager.service
+if WITH_POLKIT
+ModemManager.service: ModemManager.service.polkit.in
+	$(edit) $< >$@
+else
 ModemManager.service: ModemManager.service.in
 	$(edit) $< >$@
 endif
+endif
 
 
 # DBus Activation file
diff --git a/data/ModemManager.service.polkit.in b/data/ModemManager.service.polkit.in
new file mode 100644
index 00000000..6b745ed6
--- /dev/null
+++ b/data/ModemManager.service.polkit.in
@@ -0,0 +1,22 @@
+[Unit]
+Description=Modem Manager
+Requires=polkit.service
+After=polkit.service
+
+[Service]
+Type=dbus
+BusName=org.freedesktop.ModemManager1
+ExecStart=@sbindir@/ModemManager
+StandardError=null
+Restart=on-abort
+CapabilityBoundingSet=CAP_SYS_ADMIN
+ProtectSystem=true
+ProtectHome=true
+PrivateTmp=true
+RestrictAddressFamilies=AF_NETLINK AF_UNIX
+NoNewPrivileges=true
+User=root
+
+[Install]
+WantedBy=multi-user.target
+Alias=dbus-org.freedesktop.ModemManager1.service
-- 
2.17.1

